R/nm-join.R

```diff
@@ -109,6 +109,12 @@ nm_join <- function(
     stop(glue("couldn't find `.join_col` {.join_col} in data with cols: {paste(names(.d), collapse = ', ')}"))
   }
 
+  if(.d[.join_col] %>% anyDuplicated() != 0)
```

Comment:
Just an observation that the `%>%` with `!=0` seems especially confusing here. 

```r
if(anyDuplicated(.d[.join_col]))
```

feels much cleaner and more expressive. You see this sort of idiom [all over the R code base](https://github.com/wch/r-source/search?q=anyDuplicated&type=code) too. 

I realize your style involves pipes and don't want to change your style. Just reflecting on the idiom and readability.

Reply:
@kylebaron Yeah I agree with your 3 suggestions, especially regarding 1000's of duplicates. Think its safe to just drop the dups column. Ill make a PR to address these things and mark it for this milestone